### PR TITLE
Sound reactive on esp8266

### PR DESCRIPTION
**Pull request by FrancoisAltieri**
PR to enable sound reactive on esp82.
Not compatible with FFT effects (unless streamed from other sources)
Open question: maybe it could be done by implementing another audio source rather than compilation flags?
Disclaimer: I used the original MM esp82 sound reactive branch to know how to bring it here (see commit history, one of them is actually just bringing the old mod). I'm quite puzzled with this since MM branch was under another licensing scheme.

### CODE REVIEW

File: .devcontainer/Dockerfile
Comment (FrancoisAltieri) [inline .devcontainer/Dockerfile:1]: Unrelated, simply fixing dev container.
To be removed

File: platformio.ini
Comment (FrancoisAltieri) [inline platformio.ini:1]: Sandbox to be cleaned 
Important part is the default flags for enabling sound reactive and have needed libraries

File: wled00/fal_audio_reactive.h.delete
Comment (FrancoisAltieri) [inline wled00/fal_audio_reactive.h.delete:1]: To be removed

File: wled00/usermod.cpp.delete
Comment (FrancoisAltieri) [inline wled00/usermod.cpp.delete:1]: To be removed

File: usermods/audioreactive/audio_reactive.h [hunk 1]
```diff
@@ -2153,12 +2303,14 @@ class AudioReactive : public Usermod {
     void loop()
     {
       static unsigned long lastUMRun = millis();
+      static unsigned long lastEntry = millis();
 
-      if (!enabled) {
+      if (!enabled || millis()-lastEntry < LOOP_THROTTLING) {
```
Comment (FrancoisAltieri) [inline usermods/audioreactive/audio_reactive.h:2308]: Need to review throttling here. Running here more often is too much for esp82.

File: usermods/audioreactive/audio_reactive.h [hunk 2]
```diff
@@ -2237,7 +2389,7 @@ class AudioReactive : public Usermod {
 
         // run filters, and repeat in case of loop delays (hick-up compensation)
         if (userloopDelay <2) userloopDelay = 0;      // minor glitch, no problem
-        if (userloopDelay >200) userloopDelay = 200;  // limit number of filter re-runs  
+        if (userloopDelay >RERUNS_DELAY_CAP) userloopDelay = RERUNS_DELAY_CAP;  // limit number of filter re-runs
```
Comment (FrancoisAltieri) [inline usermods/audioreactive/audio_reactive.h:2392]: Also limiting number of iterations here. Most likely, it can also be cleaned up

**Comment by netmindz** [comment]
We are just about to remove **all** support for 8266 from the next release of WLED-MM, so this PR will not be accepted

**Comment by ewowi** [comment]
🎉

**Comment by DedeHai** [comment]
@FrancoisAltieri  thx for the code, unfortunately this is more of a hack than a proper implementation, I was hoping you had it done by adding a proper audio source in audio_source.h

**Comment by FrancoisAltieri** [comment]
> @FrancoisAltieri thx for the code, unfortunately this is more of a hack than a proper implementation, I was hoping you had it done by adding a proper audio source in audio_source.h

Indeed, first time in this code so I learned as I went and realized eventually that a new audio source would be a much cleaner solution. But since support for this is dropped I don't think I'll work on this refactoring

**Pull request closed without merging**